backend/utils.py

```diff
@@ -136,5 +136,61 @@ def format_stream_response(chatCompletionChunk, history_metadata, message_uuid=N
                     }
                     response_obj["choices"][0]["messages"].append(messageObj)
                     return response_obj
-    
-    return {}
\ No newline at end of file
+
+    return {}
+
+
+def format_pf_non_streaming_response(
+    chatCompletion, history_metadata, response_field_name, message_uuid=None
+):
+    if chatCompletion is None:
+        logging.error(
+            "chatCompletion object is None - Increase PROMPTFLOW_RESPONSE_TIMEOUT parameter"
+        )
+        return {
+            "error": "No response received from promptflow endpoint increase PROMPTFLOW_RESPONSE_TIMEOUT parameter or check the promptflow endpoint."
+        }
+    if "error" in chatCompletion:
+        logging.error(f"Error in promptflow response api: {chatCompletion['error']}")
+        return {"error": chatCompletion["error"]}
+
+    logging.debug(f"chatCompletion: {chatCompletion}")
+    try:
+        response_obj = {
+            "id": chatCompletion["id"],
+            "model": "",
+            "created": "",
+            "object": "",
+            "choices": [
+                {
+                    "messages": [
+                        {
+                            "role": "assistant",
+                            "content": chatCompletion[response_field_name],
+                        }
+                    ]
+                }
+            ],
+            "history_metadata": history_metadata,
+        }
+        return response_obj
+    except Exception as e:
+        logging.error(f"Exception in format_pf_non_streaming_response: {e}")
+        return {}
+
+
+def convert_to_pf_format(input_json, request_field_name, response_field_name):
+    output_json = []
+    # align the input json to the format expected by promptflow chat flow
+    for message in input_json["messages"]:
+        if message:
+            if message["role"] == "user":
+                new_obj = {
+                    "inputs": {request_field_name: message["content"]},
+                    "outputs": {response_field_name: ""},
+                }
+                output_json.append(new_obj)
+            elif message["role"] == "assistant" and output_json:
```

Comment:
Can you explain why this part of the condition is needed?  I also commented on this in the previous PR, because I think it will always be true.

Reply:
For clarification, the part I'm asking about is the value of `output_json`, which should always be a list based one what I'm seeing above, and therefore will always evaluate to True since it isn't None or False.

Are you maybe instead interested in the length of `output_json`, since you are accessing its last element below?

Reply:
Yes, i am trying to avoid IndexError when accessing output_json[-1] if output_json is empty list.  i can instead check for length also.

Reply:
Actually, you can disregard this comment.  I didn't realize empty list evaluated as False until I just tried it.  😄  I learned something new today.